### ISSUE TITLE
[4.x]: SQL error on searchindex job: Data too long for column 'attribute'

### ISSUE THREAD

**Issue by rauwebieten** [craftcms/cms#15113]
### Description

SQL error on searchindex job

```
yii\\db\\Exception(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'attribute' at row 1
The SQL being executed was: INSERT INTO `searchindex` (`elementId`, `attribute`, `fieldId`, `siteId`, `keywords`) VALUES (348074,'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 0, 3, '') at /var/www/ourwebsite/production/vendor/yiisoft/yii2/db/Schema.php:676)
[previous exception] [object] (PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated:
 1406 Data too long for column 'attribute' at row 1 at /var/www/ourwebsite/production/vendor/yiisoft/yii2/db/Command.php:1302)
```

The string "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" (not the actual string) is a handle for a form field in a form made with the freeform plugin. (Our client made a form and chose a rather long formfield handle)

The attribute data is too long for the varchar 25 column.

### Craft CMS version

Craft Pro 4.8.1

### PHP version

8.1.28

### Operating system and version

Linux 4.19.0-26-amd64

### Database type and version

MariaDB 10.3.39

### Image driver and version

GD 8.1.28

### Installed plugins and versions

Freeform 5.1.2

**Comment by i-just** [comment]
Hi, thanks for getting in touch! The value for the `attribute` column for custom fields (including a Freeform Form field) is, by default, “field”. Do you have any custom modules or plugins that influence the search indexing behaviour? Are you able to share a full stack trace leading up to this error?

**Comment by brandonkelly** [comment]
This sounds related to a Freeform bug I was helping with a month ago. Checking with them…

**Comment by brandonkelly** [comment]
Yep, this was fixed in Freeform 5.1.17: https://github.com/solspace/craft-freeform/blob/v5/CHANGELOG.md#5117---2024-05-03
